__tests__/unit-tests/List-Of-Friends-test.tsx

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import { render, fireEvent } from "@testing-library/react-native";
+import ListOfFriends from "@/components/friends/ListOfFriends";
+
+describe("ListOfFriends", () => {
+  const mockFriends = [
+    { uid: "1", name: "John Doe", image_id: "https://example.com/avatar1.png" },
+    { uid: "2", name: "Jane Smith", image_id: null },
+  ];
+
+  const mockHandleFriendPress = jest.fn();
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("renders a list of friends", () => {
+    const { getByText, getByTestId } = render(
+      <ListOfFriends
+        friends={mockFriends}
+        handleFriendPress={mockHandleFriendPress}
+      />,
+    );
+
+    expect(getByText("John Doe")).toBeTruthy();
+    expect(getByText("Jane Smith")).toBeTruthy();
+  });
+
+  it("displays a message if no friends are available", () => {
+    const { getByText } = render(
+      <ListOfFriends friends={[]} handleFriendPress={mockHandleFriendPress} />,
+    );
+
+    expect(getByText("You don't have any friends yet")).toBeTruthy();
+  });
+
+  it("handles friend press action", () => {
+    const { getByTestId } = render(
+      <ListOfFriends
+        friends={mockFriends}
+        handleFriendPress={mockHandleFriendPress}
+      />,
+    );
+
+    const friendItem = getByTestId("friend-item-John Doe");
```

Comment:
does this have to use the user name? Why not an index instead?

Reply:
Yes why not